File: src/methodology.js
import './methodology.css'

export function methodologyHTML() {
  return `
<section class="panel metodologia" id="metodologia">
  <h2>Como este site funciona (explicação simples)</h2>
  <p class="meto-lead">Pense em vários amigos medindo a altura da mesma árvore. Cada um usa uma fita diferente. A gente não escolhe o amigo mais alto de voz. A gente junta as medidas.</p>

  <section class="meto-sec">
    <h3>Regras de inclusão e datas</h3>
    <ul>
      <li><strong>Data de campo</strong> = quando as entrevistas foram realizadas e é a data usada no eixo do gráfico.</li>
      <li><strong>Data de publicação</strong> = quando uma fonte divulgou o levantamento. Ela permanece visível e pode ter mais de uma data de cobertura, mas não cria uma nova pesquisa.</li>
      <li><strong>Registro TSE</strong> = quando identificado, ajuda a reconhecer a mesma pesquisa mesmo que outra fonte a publique depois.</li>
      <li><strong>Gráfico nacional</strong> = somente pesquisas verificadas de intenção de voto presidencial no Brasil. Pesquisas estaduais ficam no painel regional.</li>
      <li><strong>Faixa de incerteza</strong> = estimativa do agregado. Não é a margem de erro de uma pesquisa e não é probabilidade de vitória.</li>
    </ul>
  </section>

  <section class="meto-sec">
    <h3>O desenho</h3>
    <ul>
      <li><strong>Bolinha</strong> = uma pesquisa. É só aquela escola, naquele dia.</li>
      <li><strong>Linha cheia</strong> = o resumo de várias bolinhas. É o número do site.</li>
      <li><strong>Linha tracejada</strong> = “se continuar assim”. Não é o resultado da eleição.</li>
      <li><strong>Linhas extras (overlays)</strong> = réguas em cima do resumo. Elas não mudam o número dos cartões.</li>
    </ul>
    <p class="ex">Exemplo: Datafolha diz 39 e Atlas diz 43 no mesmo fim de semana. As duas bolinhas aparecem. A linha fica no meio, um pouco mais perto de quem ouviu mais gente.</p>
  </section>

  <section class="meto-sec">
    <h3>Por que não olhamos só uma pesquisa</h3>
    <ul>
      <li>Um amigo sempre mede um pouco alto. Outro sempre mede um pouco baixo.</li>
      <li>Quem ouviu mais pessoas (N grande) pesa mais, mas não manda sozinho.</li>
      <li>Medida velha vale menos que medida nova.</li>
      <li>Se o mesmo amigo manda três recados na mesma semana, cada recado vale menos — senão ele grita mais alto só porque falou três vezes.</li>
    </ul>
    <p class="ex">Exemplo: Atlas ouve 5.000 pessoas. Datafolha ouve 2.000. Atlas pesa mais, mas Datafolha, Quaest e Nexus ainda puxam a linha. Uma casa não “ganha” o gráfico.</p>
  </section>

  <section class="meto-sec">
    <h3>Os modos da linha (Modelos 1 a 5)</h3>
    <p>É a mesma turma de bolinhas. Muda só o jeito de fazer a média. Um modo de cada vez.</p>
    <div class="meto-models">
      <article>
        <h4>off</h4>
        <p>Bolinha + linha antiga. Sem “se continuar assim”.</p>
      </article>
      <article>
        <h4>1 — padrão</h4>
        <p>Receita antiga do site. Pesquisa grande e nova pesa mais. Não corrige o hábito da casa.</p>
        <p class="ex">Exemplo: 39 ontem e 35 há 20 dias. O 39 puxa mais a linha.</p>
      </article>
      <article>
        <h4>2 — casa justa</h4>
        <p>Se uma casa sempre fica 2 pontos acima das outras, a gente tira esses 2 antes de misturar.</p>
        <p class="ex">Exemplo: casa A sempre 2 pontos acima de B, C e D. No modo 2 a linha trata A como se fosse “menos 2”.</p>
      </article>
      <article>
        <h4>3 — quando brigam</h4>
        <p>Olha a margem de erro e o quanto as casas discordam. Se discordam muito, ninguém manda sozinho.</p>
        <p class="ex">Exemplo: uma diz 34 e outra 40. O modo 3 não finge que são a mesma foto. A linha fica no meio e não corre para o 40.</p>
      </article>
      <article>
        <h4>4 — filme, não foto</h4>
        <p>A disputa anda um pouquinho por dia. Cada pesquisa nova empurra o filme, não apaga o capítulo anterior.</p>
        <p class="ex">Exemplo: três semanas de 38 e uma pesquisa de 43. O filme sobe, mas não pula para 43 no mesmo dia.</p>
      </article>
      <article>
        <h4>5 — nervoso</h4>
        <p>Olha quase só o que acabou de sair. Bom para ver o susto. Ruim para um número calmo.</p>
        <p class="ex">Exemplo: sai um 43 hoje. O modo 5 sobe rápido. O modo 1 sobe devagar.</p>
      </article>
    </div>
  </section>

  <section class="meto-sec">
    <h3>Réguas (overlays / indicadores)</h3>
    <p>São adesivos no desenho. Ligar ou desligar não muda os cartões.</p>
    <ul>
      <li><strong>SMA 7</strong> — média dos últimos 7 dias da linha. Como a nota da semana.</li>
      <li><strong>SMA 21</strong> — média de 21 dias. Como a nota do mês. Mais lisa.</li>
      <li><strong>EMA 9 / 21</strong> — parecida com a SMA, mas o dia de ontem pesa mais que o de três semanas.</li>
      <li><strong>HMA</strong> — régua rápida. Vira cedo quando a linha vira.</li>
      <li><strong>VWMA</strong> — dias com pesquisa grande (muito N) puxam mais a régua.</li>
      <li><strong>KAMA</strong> — anda rápido quando o número está mudando de verdade; anda devagar quando só treme.</li>
      <li><strong>Bollinger (BB)</strong> — um corredor em volta da linha. Corredor largo = as casas não combinam. Estreito = quase o mesmo número.</li>
    </ul>
    <p class="ex">Exemplo: a linha sobe de 36 para 39. SMA 21 quase não se mexe. EMA 9 e HMA sobem junto. BB fica mais largo se uma casa ficou em 34 e outra em 43.</p>
  </section>

  <section class="meto-sec">
    <h3>Escalas e botões do gráfico</h3>
    <ul>
      <li><strong>Eixo de baixo (X)</strong> = calendário. Esquerda é passado. Direita é agora.</li>
      <li><strong>Eixo de lado (Y)</strong> = porcentagem de voto. 40 quer dizer 40 em cada 100 pessoas naquela pesquisa.</li>
      <li><strong>30d / 90d / tudo</strong> = quanto calendário cabe na tela. Não apaga pesquisa; só aproxima o zoom.</li>
      <li><strong>Janela 7d / 14d / 30d</strong> = até que distância uma bolinha ainda puxa a linha de hoje. Janela curta = memória curta.</li>
      <li><strong>Resetar eixos</strong> = volta o zoom. <strong>Resetar Y</strong> = só o eixo da porcentagem.</li>
      <li><strong>Roda do mouse / pinça / Shift+arrastar</strong> = zoom. Como aproximar um mapa.</li>
      <li><strong>1º turno / 2º turno</strong> = duas perguntas diferentes. Não misturamos as bolinhas.</li>
    </ul>
    <p class="ex">Exemplo: no 1º turno Lula pode ter 39 e Flávio 35. No 2º a mesma pesquisa pode ser 46 a 44. São contas separadas. Trocar o chip de turno é trocar de caderno.</p>
  </section>

  <section class="meto-sec">
    <h3>Ferramentas da página</h3>
    <ul>
      <li><strong>Cartões em cima</strong> = o número da linha hoje e a diferença contra 30 dias atrás. “+1,20 pp vs 30d” = subiu um pouco no mês.</li>
      <li><strong>Chips de instituto</strong> = ligar/desligar uma casa. Útil para ver se uma casa sozinha puxa o desenho. Com todas ligadas o site está no modo justo.</li>
      <li><strong>Tema / bandeira</strong> = só cor da página. Zero efeito no número.</li>
      <li><strong>Verificar agora</strong> = pede de novo o arquivo que já está no ar. Não sai caçando pesquisa nova na hora.</li>
      <li><strong>Tabela</strong> = lista crua: quem mediu, quantas pessoas, margem, link.</li>
    </ul>
    <p class="ex">Exemplo: desliga Atlas e a linha desce um pouco. Liga de novo e ela volta. Isso mostra o peso daquela casa — por isso o modo justo deixa todas ligadas.</p>
  </section>

  <section class="meto-sec">
    <h3>O que o site não faz</h3>
    <ul>
      <li>Não diz “vai ganhar”. Diz “as pesquisas de agora, juntas, estão aqui”.</li>
      <li>Não inventa pesquisa. Sem número publicado, não tem bolinha.</li>
      <li>Não usa pesquisa de um só estado no desenho nacional.</li>
      <li>Não transforma 1º turno em 2º turno por mágica.</li>
    </ul>
  </section>

  <details class="meto-more">
    <summary>Contas diretas (para quem quiser o detalhe)</summary>
    <ul>
      <li>Modelo 1: <code>peso = √(N/2000) × exp(−dias/janela)</code>. N vazio = 800; N entre 100 e 8000.</li>
      <li>Modelo 2: meia-vida <code>2^(−dias/janela)</code>, vezes <code>1/k</code> se a mesma casa repetiu, menos o vício da casa (house) encolhido <code>n/(n+4)</code>.</li>
      <li>Modelo 3: <code>peso = recência / (σ² + τ²) / k</code>. σ = margem/1,96. τ² = briga extra entre casas.</li>
      <li>Modelo 4: Kalman + smoother. A disputa é um estado que anda ~0,16 ponto por raiz de dia.</li>
      <li>Modelo 5: meia-vida = janela/5, com impulso <code>1 + 2e^(−dias/1,8)</code> nas pesquisas novas.</li>
    </ul>
    <p>Texto longo: <a href="https://github.com/Gitdisd/pesquisas-eleitorais-br/blob/main/docs/MODELS.md" target="_blank" rel="noopener">docs/MODELS.md</a>.</p>
  </details>

  <p class="meto-foot">Site estático, sem fins partidários.</p>
  <p id="projMethodology"></p>
</section>`
}

File: rust/web-ui/src/methodology.rs
use dioxus::prelude::*;

pub fn Methodology(language: crate::ui::Language) -> Element {
    let en = matches!(language, crate::ui::Language::En);
    rsx! {
        section { class: "panel metodologia", id: "methodology",
            h2 { if en { "How this site works" } else { "Como este site funciona (explicação simples)" } }
            p { class: "meto-lead",
                if en {
                    "Think of several friends measuring the same tree. Each uses a different tape. We combine the measurements instead of choosing one voice."
                } else {
                    "Pense em vários amigos medindo a altura da mesma árvore. Cada um usa uma fita diferente. A gente junta as medidas."
                }
            }

            section { class: "meto-sec",
                h3 { if en { "Inclusion rules and dates" } else { "Regras de inclusão e datas" } }
                ul {
                    li { strong { if en { "Fieldwork date" } else { "Data de campo" } } " — "
                        if en { "when interviews were conducted and the date used on the chart axis." }
                        else { "quando as entrevistas foram realizadas e é a data usada no eixo do gráfico." }
                    }
                    li { strong { if en { "Publication date" } else { "Data de publicação" } } " — "
                        if en { "when the source released the poll. It remains visible and does not create a second poll." }
                        else { "quando uma fonte divulgou o levantamento. Ela permanece visível e não cria uma nova pesquisa." }
                    }
                    li { strong { if en { "TSE registration" } else { "Registro TSE" } } " — "
                        if en { "when identified, it helps recognize the same poll when published by another source." }
                        else { "quando identificado, ajuda a reconhecer a mesma pesquisa mesmo que outra fonte a publique depois." }
                    }
                    li { strong { if en { "National chart" } else { "Gráfico nacional" } } " — "
                        if en { "only verified presidential voting-intention polls in Brazil. State polls stay in the regional panel." }
                        else { "somente pesquisas verificadas de intenção de voto presidencial no Brasil. Pesquisas estaduais ficam no painel regional." }
                    }
                    li { strong { if en { "Uncertainty band" } else { "Faixa de incerteza" } } " — "
                        if en { "an estimate around the aggregate. It is not a poll margin of error or a win probability." }
                        else { "uma estimativa ao redor do agregado. Não é a margem de erro de uma pesquisa e não é probabilidade de vitória." }
                }
            }

            section { class: "meto-sec",
                h3 { if en { "What the drawing means" } else { "O desenho" } }
                ul {
                    li { strong { "●" } " — " if en { "one poll, from one pollster, at that time." } else { "uma pesquisa, daquela casa, naquele período." } }
                    li { strong { if en { "Solid line" } else { "Linha cheia" } } " — " if en { "the aggregate of the points." } else { "o resumo das bolinhas." } }
                    li { strong { if en { "Dashed line" } else { "Linha tracejada" } } " — " if en { "a limited model extrapolation, not an election result." } else { "uma extrapolação limitada do modelo, não o resultado da eleição." } }
                    li { strong { if en { "Overlays" } else { "Linhas extras (overlays)" } } " — " if en { "visual reference lines that do not change card values." } else { "réguas visuais que não mudam os valores dos cartões." } }
                }
            }

            section { class: "meto-sec",
                h3 { if en { "Why combine polls" } else { "Por que não olhamos só uma pesquisa" } }
                ul {
                    li { if en { "Pollsters can systematically read somewhat high or low." } else { "Um instituto pode medir um pouco alto. Outro pode medir um pouco baixo." } }
                    li { if en { "Larger samples get more weight, but do not decide alone." } else { "Quem ouviu mais pessoas pesa mais, mas não manda sozinho." } }
                    li { if en { "Older measurements count less than newer ones." } else { "Medida velha vale menos que medida nova." } }
                    li { if en { "Repeated polls from the same pollster are down-weighted so repetition alone does not dominate." } else { "Pesquisas repetidas da mesma casa recebem menos peso para que repetir não domine o desenho." } }
                }
            }

            section { class: "meto-sec",
                h3 { if en { "Line models" } else { "Os modos da linha" } }
                p { if en { "The same poll points can be summarized in different ways. One mode is active at a time." } else { "É a mesma turma de bolinhas. Muda só o jeito de fazer a média. Um modo de cada vez." } }
                div { class: "meto-models",
                    for (id, title, body) in model_copy(en) {
                        article {
                            h4 { "{title}" }
                            p { "{id}" }
                            p { "{body}" }
                        }
                    }
                }
            }

            section { class: "meto-sec",
                h3 { if en { "Visual overlays" } else { "Réguas (overlays / indicadores)" } }
                p { if en { "They are visual guides over the selected trend. Turning them on or off does not change the headline cards." } else { "São réguas visuais sobre a tendência selecionada. Ligar ou desligar não muda os cartões." } }
                ul {
                    for (label, body) in overlay_copy(en) {
                        li { strong { "{label}" } " — {body}" }
                    }
                }
            }

            section { class: "meto-sec",
                h3 { if en { "Chart controls" } else { "Escalas e botões do gráfico" } }
                ul {
                    li { strong { "X" } " — " if en { "calendar: left is older, right is newer." } else { "calendário: esquerda é passado, direita é mais nova." } }
                    li { strong { "Y" } " — " if en { "vote share percentage." } else { "porcentagem de intenção de voto." } }
                    li { strong { "1d / 3d / 7d / 14d / 21d / 30d / 90d / all" } " — " if en { "how much calendar is shown; it does not delete polls." } else { "quanto calendário aparece; não apaga pesquisas." } }
                    li { strong { "7d / 14d / 30d / 90d / YTD / custom" } " — " if en { "how far back the selected trend reaches." } else { "até que distância a janela da tendência alcança." } }
                    li { strong { "Wheel / pinch / drag" } " — " if en { "native SVG navigation." } else { "navegação nativa do SVG." } }
                }
            }

            section { class: "meto-sec",
                h3 { if en { "Page tools" } else { "Ferramentas da página" } }
                ul {
                    li { strong { if en { "Candidate cards" } else { "Cartões" } } " — " if en { "current selected-model estimate and 30-day difference." } else { "estimativa do modelo selecionado e diferença contra 30 dias." } }
                    li { strong { if en { "Pollster filters" } else { "Filtros de instituto" } } " — " if en { "include or exclude a pollster from the main chart and table." } else { "ligar ou desligar uma casa do gráfico e da tabela." } }
                    li { strong { if en { "Theme" } else { "Tema" } } " — " if en { "visual only; it has no numerical effect." } else { "somente visual; não afeta os números." } }
                    li { strong { if en { "Refresh" } else { "Atualizar" } } " — " if en { "reloads the published data already served by Pages." } else { "recarrega os dados publicados que já estão no ar." } }
                    li { strong { if en { "Table" } else { "Tabela" } } " — " if en { "lists raw poll metadata, sample, margin and source." } else { "lista metadados, amostra, margem e fonte da pesquisa." } }
                }
            }

            details { class: "meto-more",
                summary { if en { "Direct formulas" } else { "Contas diretas (para quem quiser o detalhe)" } }
                ul {
                    li { "Model 1: " code { "weight = √(N/2000) × exp(−days/window)" } }
                    li { "Model 2: " code { "2^(−days/window), repeated-house down-weighting, house effect correction" } }
                    li { "Models 3–12 use the shared Rust advanced-model dispatcher and the same production parity fixtures." }
                }
                p {
                    if en { "Long-form documentation: " }
                    else { "Texto longo: " }
                    a { href: "https://github.com/Gitdisd/pesquisas-eleitorais-br/blob/main/docs/MODELS.md", target: "_blank", rel: "noopener noreferrer", "docs/MODELS.md" }
                }
            }

            p { class: "meto-foot",
                if en { "Static, non-partisan site." } else { "Site estático, sem fins partidários." }
            }
        }
    }
}

fn model_copy(en: bool) -> Vec<(&'static str, &'static str, &'static str)> {
    if en {
        vec![
            ("0", "off", "Individual points without model extrapolation."),
            ("1", "standard", "Large and recent polls weigh more using the canonical production rule."),
            ("2", "house-adjusted", "Adds pollster house-effect adjustment and repeated-house down-weighting."),
            ("3", "Meta", "Accounts for sampling uncertainty and disagreement between pollsters."),
            ("4", "Kalman", "Treats latent intent as a state updated by incoming polls."),
            ("5", "Fast", "Uses a shorter memory and stronger emphasis on recent measurements."),
        ]
    } else {
        vec![
            ("0", "off", "Bolinha + linha sem extrapolação."),
            ("1", "padrão", "Pesquisa grande e nova pesa mais pela regra canônica de produção."),
            ("2", "casa justa", "Inclui ajuste de efeito de casa e redução de repetição do mesmo instituto."),
            ("3", "Meta", "Considera incerteza amostral e discordância entre institutos."),
            ("4", "Kalman", "Trata a intenção latente como um estado atualizado por novas pesquisas."),
            ("5", "Rápido", "Usa memória curta e mais ênfase nas medições recentes."),
        ]
    }
}

fn overlay_copy(en: bool) -> Vec<(&'static str, &'static str)> {
    if en {
        vec![
            ("SMA 7", "simple 7-point moving average."),
            ("SMA 21", "simple 21-point moving average."),
            ("EMA 9 / 21", "exponential moving averages that emphasize newer points."),
            ("HMA 16", "faster weighted moving average."),
            ("VWMA 14", "volume/sample-size weighted moving average."),
            ("KAMA 10", "adaptive moving average that reacts to efficiency in the series."),
            ("Bollinger 20", "a midline with a ±2 standard-deviation visual corridor."),
        ]
    } else {
        vec![
            ("SMA 7", "média móvel simples de 7 pontos."),
            ("SMA 21", "média móvel simples de 21 pontos."),
            ("EMA 9 / 21", "médias exponenciais que dão mais peso aos pontos novos."),
            ("HMA 16", "média móvel ponderada mais rápida."),
            ("VWMA 14", "média móvel ponderada pelo tamanho da amostra."),
            ("KAMA 10", "média adaptativa que reage à eficiência da série."),
            ("Bollinger 20", "linha central com corredor visual de ±2 desvios-padrão."),
        ]
    }
}
